### DISCUSSION TITLE
Sonoff Tx1c v2.0.0

### DISCUSSION THREAD

**Discussion by mtaylor819** [arendst/Tasmota#15057] [Q&A] [answered]
Has anyone seen the new versions of TX1 switches for EU / UK ? I can't get the board into flashing mode ready to upload ? 

I thought I had found GPIO 0 but not convinced !

I have added some pics for reference.
![processed-a6d36aa8-ba3f-459b-9355-4acec3f920cc_1MWliwrJ](https://user-images.githubusercontent.com/16101312/157070809-282fd885-a4d5-4097-8e73-3a99a1163e9e.jpeg)
![processed-d4b3d1e3-1e02-4b3f-9be6-86ff1aaaec7e_7ajhhK3t](https://user-images.githubusercontent.com/16101312/157070817-a16ae0e4-2f13-4cb5-a253-c7413532df10.jpeg)
![processed-9ea15557-e782-43be-8bf7-58df54003c43_9gYKPKWv](https://user-images.githubusercontent.com/16101312/157070826-576d7298-d3a6-46e0-977a-564a5e852a10.jpeg)
![processed-c8c04c35-c927-41f5-8234-9993a0c4ab5a_Bk2FTucD](https://user-images.githubusercontent.com/16101312/157070834-03c2c2d3-3c3f-47c2-a577-9f6ea6fa7a6f.jpeg)

**Comment by ascillato2** [comment, marked as answer]
For finding GPIO0, the explanations on https://github.com/arendst/Tasmota/discussions/15052 can be of help to you too.

**Reply by sfromis** [reply]
Instead, I'd recommend to just not hunt for where GPIO0 is "hidden" on some iteration of those switches. If you check the [pinout](https://tasmota.github.io/docs/Pinouts/), you'll see that gpio0 is the 2nd pin from the right on the bottom row on the ESP chips. As the next is GPIO4, which you can safely ground while flashing, you can simply push a ground-connected pin between GPIO0 and GPIO4 when applying power. Not at all hard to do, and works for all those Sonoff Tx switches.

**Reply by Panapat1** [reply]
TP1 Key0 is GPIO0 for this board, not R19 or R18

**Comment by hezi-pereg** [comment]
Hi, If it is still relevant to anyone, I managed to do it. 
Please note that GPIO0 is represented in TP1 KEY0 (as in the attached picture):
![157070809-282fd885-a4d5-4097-8e73-3a99a1163e9e](https://user-images.githubusercontent.com/98700156/163966240-a060048c-ebe6-47a5-b6ce-d1c9f8225e94.jpeg)
Hope it helps

**Reply by heisenberg2980** [reply]
Did you use Tasmotizer to flash it or a different tool? I am trying to flash one of those and getting an error, but I believe the connections are correct as the computer detects the port:
![image](https://user-images.githubusercontent.com/58039006/213285632-67530ab7-2e92-4349-b511-6d4cf2b0272a.png)

**Reply by sfromis** [reply]
Usual reason for that is not grounding gpio0, bad connections or insufficient 3.3V power available. More than a few USB/serial adapters are not up to the task of acting as a power supply during flashing. Getting the port detected tells nothing more than USB plugged in.
